Jellyfin.Api/Controllers/MediaSegmentController.cs

```diff
@@ -0,0 +1,140 @@
+using MediaBrowser.Model.MediaSegments;
+using MediaBrowser.Model.Querying;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Jellyfin.Api.Controllers;
+
+/// <summary>
+/// Media Segments controller.
+/// </summary>
+[Authorize]
+public class MediaSegmentController : BaseJellyfinApiController
+{
+    private readonly IMediaSegmentsManager _mediaSegmentManager;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="MediaSegmentController"/> class.
+    /// </summary>
+    /// <param name="mediaSegmentManager">Instance of the <see cref="IMediaSegmentsManager"/> interface.</param>
+    public MediaSegmentController(
+        IMediaSegmentsManager mediaSegmentManager)
+    {
+        _mediaSegmentManager = mediaSegmentManager;
+    }
+
+    /// <summary>
+    /// Get all media segments.
+    /// </summary>
+    /// <param name="itemId">Optional: Just segments with MediaSourceId.</param>
+    /// <param name="streamIndex">Optional: Just segments with MediaStreamIndex.</param>
+    /// <param name="type">Optional: All segments of type.</param>
+    /// <param name="typeIndex">Optional: All segments with typeIndex.</param>
+    /// <response code="200">Segments returned.</response>
+    /// <returns>An <see cref="OkResult"/>containing the queryresult of segments.</returns>
+    [HttpGet]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    public ActionResult<QueryResult<MediaSegment>> GetSegments(
+        [FromQuery] Guid itemId,
+        [FromQuery, DefaultValue(-1)] int streamIndex,
+        [FromQuery] MediaSegmentType? type,
+        [FromQuery, DefaultValue(-1)] int typeIndex)
+    {
+        var list = _mediaSegmentManager.GetAllMediaSegments(itemId, streamIndex, typeIndex, type);
+
+        return new QueryResult<MediaSegment>(list);
+    }
+
+    /// <summary>
+    /// Create or update a media segment. You can update start/end/action.
+    /// </summary>
+    /// <param name="startTicks">Start position of segment in Ticks.</param>
+    /// <param name="endTicks">End position of segment in Ticks.</param>
+    /// <param name="itemId">Segment is associated with MediaSourceId.</param>
+    /// <param name="streamIndex">Segment is associated with MediaStreamIndex.</param>
+    /// <param name="type">Segment type.</param>
+    /// <param name="typeIndex">Optional: If you want to add a type multiple times to the same itemId increment it.</param>
+    /// <param name="action">Optional: Creator recommends an action.</param>
+    /// <response code="200">Segments returned.</response>
+    /// <response code="400">Missing query parameter.</response>
+    /// <returns>An <see cref="OkResult"/>containing the queryresult of segment.</returns>
+    [HttpPost("Segment")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    public async Task<ActionResult<QueryResult<MediaSegment>>> PostSegment(
```

Comment:
Do we really want users (non admins) to have access to this api?

Reply:
I'd argue that creating and modifying segments should either be behind a permission or admin only.

Reply:
Indeed it shouldn't be possible without admin or api key. The same for deletion.